### PR TITLE
feat: add croc spec

### PR DESCRIPTION
**Pull request by brandonleon**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Feature, adding spec for croc, the command line tool for sending files and folders.

**What is the current behavior? (You can also link to an open issue here)**
No current spec for croc.

**What is the new behavior (if this is a feature change)?**
Adding spec for the croc command.

**Additional info:**

### CODE REVIEW

File: src/croc.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    "Croc is a tool that allows any two computers to simply and securely transfer files and folders",
+  subcommands: [
+    {
+      name: "send",
+      description: "Send file(s), or folder",
+    },
+    {
+      name: "relay",
+      description: "Start your own relay",
+    },
```
Comment (SeparateRecords) [inline src/croc.ts:13]: `send` and `relay` both have options but it looks like they're missing here

File: src/croc.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+      name: "--throttleUpload",
+      description: "Throttle the upload speed e.g. 500k",
+    },
+    {
+      name: ["--version", "-v"],
+      description: "Print the version",
+    },
+  ],
+  // Only uncomment if croc takes an argument
+  // args: {}
```
Comment (SeparateRecords) [inline src/croc.ts:108]: It should take an argument which is the secret code

**Review by SeparateRecords — CHANGES_REQUESTED** [review]
Thanks for the PR! Left some notes - not much left to add 🙂

File: src/croc.ts (hunk 2 above)
Comment (brandonleon) [reply src/croc.ts]: Added receive code argument.

File: src/croc.ts [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+      args: {
+        name: "command",
+        template: "help",
+      },
+    },
+  ],
+  options: [
+    {
+      name: ["--help", "-h"],
+      description: "Show help for croc",
```
Comment (SeparateRecords) [inline src/croc.ts:76]:
```suggestion
      description: "Show help for croc",
      isPersistent: true,
```

File: src/croc.ts [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+      description: 'Password for the relay (default: "pass123")',
+    },
+    {
+      name: "--socks5",
+      description: "Add a socks5 proxy",
+    },
+    {
+      name: "--throttleUpload",
+      description: "Throttle the upload speed e.g. 500k",
+    },
```
Comment (SeparateRecords) [inline src/croc.ts:159]: These should all take an argument, `args: { name: "value" }` on each of them would be good enough!

File: src/croc.ts [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+      description: "Start your own relay",
+      options: [
+        {
+          name: "--host",
+          description: "Host of the relay",
+        },
+        {
+          name: "--ports",
+          description: "Ports of the local relay",
+        },
```
Comment (SeparateRecords) [inline src/croc.ts:61]: These both take a value, can just do `args: { name: "value" }`

**Review by SeparateRecords — CHANGES_REQUESTED** [review]
Done a final pass on it, checking for missing args. I've marked the options that should take an arg, once that's done it's ready to merge!

File: src/croc.ts (hunk 4 above)
Comment (brandonleon) [reply src/croc.ts:160]: Got a little more detailed on a few of them.

**Review by SeparateRecords — APPROVED** [review]
LGTM! Thanks for adding all of that 😄

Note that you probably won't instantly get this spec since the package size is now over 100MB, which causes issues for the CDN we use (jsdelivr).

I have an issue open to increase the size limit for us and have a workaround PR up internally, so hopefully it'll be usable soon!

**Pull request merged**